packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -9,114 +9,234 @@
+    });
+    Object.defineProperty(event, 'returnValue', {
+      get() {
+        return false;
+      },
+    });
+    node.dispatchEvent(event);
+
+    expect(expectedCount).toBe(2);
   });
 
   it('should be able to `stopPropagation`', () => {
-    var nativeEvent = {};
-    var syntheticEvent = createEvent(nativeEvent);
+    var node;
+    var expectedCount = 0;
 
-    expect(syntheticEvent.isPropagationStopped()).toBe(false);
-    syntheticEvent.stopPropagation();
-    expect(syntheticEvent.isPropagationStopped()).toBe(true);
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isPropagationStopped()).toBe(false);
+      syntheticEvent.stopPropagation();
+      expect(syntheticEvent.isPropagationStopped()).toBe(true);
 
-    expect(nativeEvent.cancelBubble).toBe(true);
+      expectedCount++;
+    };
+    node = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    node.dispatchEvent(event);
+
+    expect(expectedCount).toBe(1);
   });
 
   it('should be able to `persist`', () => {
-    var syntheticEvent = createEvent({});
+    var node;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      expect(syntheticEvent.isPersistent()).toBe(false);
+      syntheticEvent.persist();
+      expect(syntheticEvent.isPersistent()).toBe(true);
 
-    expect(syntheticEvent.isPersistent()).toBe(false);
-    syntheticEvent.persist();
-    expect(syntheticEvent.isPersistent()).toBe(true);
+      expectedCount++;
+    };
+    node = ReactDOM.render(<div onClick={eventHandler} />, container);
+
+    var event;
+    event = document.createEvent('Event');
+    event.initEvent('click', true, true);
+    node.dispatchEvent(event);
+
+    expect(expectedCount).toBe(1);
   });
 
   it('should be nullified if the synthetic event has called destructor and log warnings', () => {
     spyOn(console, 'error');
-    var target = document.createElement('div');
-    var syntheticEvent = createEvent({srcElement: target});
-    syntheticEvent.destructor();
-    expect(syntheticEvent.type).toBe(null);
-    expect(syntheticEvent.nativeEvent).toBe(null);
-    expect(syntheticEvent.target).toBe(null);
-    // once for each property accessed
-    expectDev(console.error.calls.count()).toBe(3);
-    // assert the first warning for accessing `type`
-    expectDev(console.error.calls.argsFor(0)[0]).toBe(
-      'Warning: This synthetic event is reused for performance reasons. If ' +
-        "you're seeing this, you're accessing the property `type` on a " +
-        'released/nullified synthetic event. This is set to null. If you must ' +
-        'keep the original synthetic event around, use event.persist(). ' +
-        'See https://fb.me/react-event-pooling for more information.',
-    );
+    var node;
+    var expectedCount = 0;
+
+    var eventHandler = syntheticEvent => {
+      syntheticEvent.destructor();
```

Comment:
This is a private API. Instead of triggering it explicitly we should save the synthetic event to a variable. Then we should assert at the end of the test (*after* dispatching events) that accessing them does not work.